--- CMakeLists.txt	2012-05-11 00:16:57.951355547 -0700
+++ CMakeLists.txt	2012-09-02 19:08:32.371155189 -0700
@@ -77,8 +77,8 @@
 add_library(tsdll SHARED ${libtsdll_SRCS})
 set_target_properties(tsdll PROPERTIES COMPILE_FLAGS "-fPIC -O3 -fno-strict-aliasing")
 
-add_executable(jconsole ${jconsole_SRCS})
-target_link_libraries(jconsole ${jconsole_LIBS})
+add_executable(jlang ${jconsole_SRCS})
+target_link_libraries(jlang ${jconsole_LIBS})
 
 if(WIN32)
     set(LIBDEST ".")
@@ -94,7 +94,7 @@
 install(TARGETS tsdll
         DESTINATION ${LIBDEST})
 
-install(TARGETS jconsole
+install(TARGETS jlang
         DESTINATION ${EXEDEST})
 
 file(GLOB j_SYSTEM
