--- GPipe.cabal	2012-06-24 15:13:24.473810871 +0200
+++ GPipe.cabal	2012-06-24 15:13:34.046269877 +0200
@@ -29,7 +29,7 @@
                     GLUT >=2.1.2.1,
                     OpenGL >=2.2.3.0,
                     Boolean == 0.0.1,
-                    Vec == 0.9.8,
+                    Vec == 0.9.9,
                     Vec-Boolean == 1.0.4,
                     base >= 4 && <5,
                     transformers
