--- src/System/INotify.hsc	2011-07-22 18:48:53.000000000 +0200
+++ ../hinotify-0.3.2-new/src/System/INotify.hsc	2012-09-15 09:43:21.283932931 +0200
@@ -38,7 +38,7 @@
 import Control.Monad
 import Control.Concurrent
 import Control.Concurrent.MVar
-import Control.Exception (bracket)
+import Control.Exception (catch, bracket, SomeException(..))
 import Data.Maybe
 import Data.Map (Map)
 import qualified Data.Map as Map
@@ -313,13 +313,13 @@
     runHandler (_,  e@QOverflow) = do -- send overflows to all handlers
         handlers <- readMVar em
         flip mapM_ (Map.elems handlers) $ \handler ->
-            catch (handler e) (\_ -> return ()) -- supress errors
+            catch (handler e) (\(SomeException _) -> return ()) -- supress errors
     runHandler (wd, event) = do 
         handlers <- readMVar em
         let handlerM = Map.lookup wd handlers
         case handlerM of
           Nothing -> putStrLn "runHandler: couldn't find handler" -- impossible?
-          Just handler -> catch (handler event) (\_ -> return ())
+          Just handler -> catch (handler event) (\(SomeException _) -> return ())
 
 killINotify :: INotify -> IO ()
 killINotify (INotify h _ _ tid1 tid2) =
