diff -ru xapian-core-1.4.0.orig/docs/doxygen_api.conf.in xapian-core-1.4.0/docs/doxygen_api.conf.in
--- xapian-core-1.4.0.orig/docs/doxygen_api.conf.in	2016-10-12 13:22:35.242574802 +0200
+++ xapian-core-1.4.0/docs/doxygen_api.conf.in	2016-10-12 13:23:47.692719046 +0200
@@ -322,14 +322,14 @@
 # various overviews, but no documentation section is generated.
 # This option has no effect if EXTRACT_ALL is enabled.
 
-HIDE_UNDOC_MEMBERS     = YES
+HIDE_UNDOC_MEMBERS     = NO
 
 # If the HIDE_UNDOC_CLASSES tag is set to YES, Doxygen will hide all
 # undocumented classes that are normally visible in the class hierarchy.
 # If set to NO (the default) these classes will be included in the various
 # overviews. This option has no effect if EXTRACT_ALL is enabled.
 
-HIDE_UNDOC_CLASSES     = YES
+HIDE_UNDOC_CLASSES     = NO
 
 # If the HIDE_FRIEND_COMPOUNDS tag is set to YES, Doxygen will hide all
 # friend (class|struct|union) declarations.
@@ -389,13 +389,13 @@
 # by member name. If set to NO (the default) the members will appear in
 # declaration order.
 
-SORT_BRIEF_DOCS        = NO
+SORT_BRIEF_DOCS        = YES
 
 # If the SORT_GROUP_NAMES tag is set to YES then doxygen will sort the
 # hierarchy of group names into alphabetical order. If set to NO (the default)
 # the group names will appear in their defined order.
 
-SORT_GROUP_NAMES       = NO
+SORT_GROUP_NAMES       = YES
 
 # If the SORT_BY_SCOPE_NAME tag is set to YES, the class list will be
 # sorted by fully-qualified names, including namespaces. If set to
diff -ru xapian-core-1.4.0.orig/docs/doxygen_source.conf.in xapian-core-1.4.0/docs/doxygen_source.conf.in
--- xapian-core-1.4.0.orig/docs/doxygen_source.conf.in	2016-10-12 13:22:35.242574802 +0200
+++ xapian-core-1.4.0/docs/doxygen_source.conf.in	2016-10-12 13:23:33.918881718 +0200
@@ -389,13 +389,13 @@
 # by member name. If set to NO (the default) the members will appear in
 # declaration order.
 
-SORT_BRIEF_DOCS        = NO
+SORT_BRIEF_DOCS        = YES
 
 # If the SORT_GROUP_NAMES tag is set to YES then doxygen will sort the
 # hierarchy of group names into alphabetical order. If set to NO (the default)
 # the group names will appear in their defined order.
 
-SORT_GROUP_NAMES       = NO
+SORT_GROUP_NAMES       = YES
 
 # If the SORT_BY_SCOPE_NAME tag is set to YES, the class list will be
 # sorted by fully-qualified names, including namespaces. If set to
